src/Nest/Ingest/Processor/ElasticClient-IngestProcessorGrok.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Elasticsearch.Net;
+
+namespace Nest
+{
+	using System.Threading;
+	using NodesHotThreadConverter = Func<IApiCallDetails, Stream, NodesHotThreadsResponse>;
+
+	public partial interface IElasticClient
+	{
+		/// <summary>
+		/// Retrieving which patterns which the grok processor is packaged with, useful as different versions are bundled with different processors.
+		/// <para> </para>https://www.elastic.co/guide/en/elasticsearch/plugins/master/ingest.html
+		/// </summary>
+		/// <param name="selector">An optional descriptor to further describe the endpoint usage operation</param>
+		IIngestProcessorGrokResponse IngestProcessorGrok(Func<IngestProcessorGrokDescriptor, IIngestProcessorGrokRequest> selector = null);
```

Comment:
Not sure about the name of this API; Perhaps `GrokProcessorPatterns`? Happy to go with consensus here

Reply:
Not sure either, just picked a name, but happy to change it.